google/cloud/pubsub/internal/backwards_compatibility_type.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BACKWARDS_COMPATIBILITY_TYPE_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BACKWARDS_COMPATIBILITY_TYPE_H
+
+#include "google/cloud/pubsub/version.h"
+#include "google/cloud/options.h"
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
```

Comment:
Do we think this type might be useful in similar circumstances in other libraries?  If so, it might be an idea to make it independent of pubsub.

Reply:
I agree that it could be useful in the future for other libraries.

My thinking was just that we could start with it in `pubsub_internal`, and only move it into `g::c::internal` if a use case in another library arises. I don't think that would constitute a breaking change, because nobody should be using the type.

I don't feel strongly either way, though.